libc/src/compiler/generic/__stack_chk_fail.cpp

```diff
@@ -12,6 +12,8 @@
 
 extern "C" {
 
+uintptr_t __stack_chk_guard;
```

Comment:
Can you assign an arbitrary random number as its default value? It should be better than zero.

Reply:
Done, hopefully the number I gave is arbitrary and random enough heh.

Reply:
I guess it is better to use `static_cast<uintptr_t>(XXX)`; otherwise a warning will be triggered on 32bit targets

Reply:
Good idea, I added that now.